src/invidious/views/watch.ecr

```diff
@@ -112,19 +112,36 @@ we're going to need to do it here in order to allow for translations.
     <div class="pure-u-1 pure-u-lg-1-5">
         <div class="h-box">
             <span id="watch-on-youtube">
-                <a href="https://www.youtube.com/watch?v=<%= video.id %>"><%= translate(locale, "videoinfo_watch_on_youTube") %></a>
-                (<a href="https://www.youtube.com/embed/<%= video.id %>"><%= translate(locale, "videoinfo_youTube_embed_link") %></a>)
+                <%-
+                    link_yt_watch = URI.new(scheme: "https", host: "www.youtube.com", path: "/watch", query: "v=#{video.id}")
+                    link_yt_embed = URI.new(scheme: "https", host: "www.youtube.com", path: "/embed/#{video.id}")
+
+                    if !plid.nil? && !continuation.nil?
+                        link_yt_param = URI::Params{"plid" => [plid], "index" => [continuation.to_s]}
+                        link_yt_watch = IV::HttpServer::Utils.add_params_to_url(link_yt_watch, link_yt_param)
+                        link_yt_embed = IV::HttpServer::Utils.add_params_to_url(link_yt_embed, link_yt_param)
+                    end
+                -%>
+                <a id="link-yt-watch" data-base-url="<%= link_yt_watch %>" href="<%= link_yt_watch %>"><%= translate(locale, "videoinfo_watch_on_youTube") %></a>
+                (<a id="link-yt-embed" data-base-url="<%= link_yt_embed %>" href="<%= link_yt_embed %>"><%= translate(locale, "videoinfo_youTube_embed_link") %></a>)
             </span>
+
             <p id="watch-on-another-invidious-instance">
-            <% if env.get("preferences").as(Preferences).automatic_instance_redirect%>
-                <a href="/redirect?referer=<%= env.get?("current_page") %>"><%= translate(locale, "Switch Invidious Instance") %></a>
-            <% else %>
-                <a href="https://redirect.invidious.io<%= env.request.resource %>"><%= translate(locale, "Switch Invidious Instance") %></a>
-            <% end %>
+                <%- link_iv_other = IV::Frontend::Misc.redirect_url(env) -%>
+                <a id="link-iv-other" data-base-url="<%= link_iv_other %>" href="<%= link_iv_other %>"><%= translate(locale, "Switch Invidious Instance") %></a>
             </p>
+
             <p id="embed-link">
-                <a href="<%= embed_link %>"><%= translate(locale, "videoinfo_invidious_embed_link") %></a>
+                <%-
+                    params_iv_embed = env.params.query.dup
+                    params_iv_embed.delete_all("v")
+
+                    link_iv_embed = URI.new(path: "/embed/#{id}")
```

Comment:
It seems like Javascript can't parse `/embed/#{id}` as valid a URL. It needs a domain for JS to parse it correctly at this line in the `addCurrentTimeToURL` function:

https://github.com/iv-org/invidious/blob/bb14f794969f62582917a39c2dd57bf92fa146a7/assets/js/player.js#L105

Reply:
Adding `window.location.origin` as the second parameter for the URL constructor, fixes the issue, while also creating a valid URL.

Reply:
Fixed, thanks for your feedback :)